web_app/model_view.py

```diff
@@ -78,17 +140,50 @@ def display_model_view(container):
 
             with selection_column:
                 corrosion_types = sorted({process_type for _, process_type in model_process_pairs})
-                selected_corrosion_type = st.selectbox('**Corrosion Type**', corrosion_types)
+                selected_corrosion_type = st.selectbox('**Corrosion Type**', corrosion_types,
+                                                       key="corrosion_type_selectbox")
 
-                filtered_models = [model for model, process_type in model_process_pairs if process_type == selected_corrosion_type]
-                selected_model = st.selectbox('**Model**', filtered_models, format_func=lambda model: model.name + " (" + model.kadi_identifier + ")")
+                filtered_models = [model for model, process_type in model_process_pairs if
+                                   process_type == selected_corrosion_type]
+                selected_model = st.selectbox(
+                    '**Model**', filtered_models,
+                    format_func=lambda model: model.name + " (" + model.kadi_identifier + ")", key="model_selectbox"
+                )
 
-                time_range = st.number_input('Enter duration [years]:', min_value=2.5, max_value=100.0, step=2.5)
+                # Capture the time range input
+                time_range = st.number_input('Enter duration [years]:', min_value=2.5, max_value=100.0, step=2.5,
+                                             key="time_range_input")
 
+                # Display parameters and compute corrosion speed and exponent using the selected time
                 selected_model.display_parameters()
 
+                # Create two columns for buttons to be side by side
+                add_button, reset_button, download_button, empty = st.columns([1, 1, 1, 3])
+                fig = generate_plot(plot_data, selected_model, time_range)
+
+                with add_button:
+                    if st.button("Add to Plot", key="add_button"):
+                        plot_data.append(selected_model)  # Add the current model to the list of models to plot
+
+                with reset_button:
+                    if st.button("Reset Plot", key="reset_button"):
+                        plot_data = []  # Reset the list of models
+
+                with download_button:
+                    if fig is not None:
+                        buffer = io.BytesIO()
+                        fig.write_image(buffer, format="png")
+                        buffer.seek(0)
+
+                        st.download_button(
```

Comment:
maybe the plot can be downloaded with 300 dpi or higher resolution. Currently, it downloads at low res and does not look nice.

Reply:
fixed in #36